Initial code design for SameSite=None known incompatible clients

tssmith04 opened the pull request:
Continuation of #15808 
Fixes #24901

mrsaldana committed:
Initial code design for SameSite=None known incompatible clients

pnicolucci commented on dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/cookies/SameSiteCookieUtils.java:
```diff
@@ -0,0 +1,272 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Copyright is incorrect, we should update to the new copyright header using EPL 2.0. We can maintain the 2021 date unless we make additional changes to this class then it would be 2021, 2024.

pnicolucci commented on dev/com.ibm.ws.transport.http/test/com/ibm/ws/http/channel/test/api/SameSiteCookieUtilsTest.java:
```diff
@@ -0,0 +1,250 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
```
Copyright is incorrect, we should update to the new copyright header using EPL 2.0. We can maintain the 2021 date unless we make additional changes to this class then it would be 2021, 2024.

pnicolucci commented on dev/com.ibm.ws.transport.http/test/com/ibm/ws/http/channel/test/api/SameSiteCookieUtilsTest.java:
```diff
@@ -0,0 +1,250 @@
+        }
+    }
+
+}
```
new line

pnicolucci commented on dev/com.ibm.ws.transport.http/test/com/ibm/ws/http/channel/test/api/SameSiteCookieUtilsTest.java:
```diff
@@ -0,0 +1,250 @@
+
+            //Test user agent on MacEmbedded browser and macOS before 10.14
+            //Test user agent on MacEmbedded browser and macOS after 10.14
+            //Test user agent on iOS before 12
```
It would be good to add tests to validate our logic if possible

tssmith04 committed:
Added Unit Tests & Remove Partitioned from Cookies with SameSiteNone

Copyright Cleanup

Remove Partitioned from Cookies with SameSiteNone

Cleanup Copyright & Comments

pnicolucci commented on dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/cookies/SameSiteCookieUtils.java:
```diff
@@ -1,9 +1,11 @@
+ * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
  *     IBM Corporation - initial API and implementation
```
can remove the contributors

pnicolucci commented on dev/com.ibm.ws.transport.http/test/com/ibm/ws/http/channel/test/api/SameSiteCookieUtilsTest.java:
```diff
@@ -1,9 +1,11 @@
+ * SPDX-License-Identifier: EPL-2.0
  *
  * Contributors:
  *     IBM Corporation - initial API and implementation
```
can remove the contributors

tssmith04 commented:
#build

pnicolucci commented:
@tssmith04 since you updated the unit test and product classes you can update the copyright date range to include 2024. We also need to look at beta guarding this functionality.

tssmith04 commented:
#build

tssmith04 commented:
Bundling all commits into one PR: #29048
